src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py

```diff
@@ -174,18 +174,22 @@ def align_timestamps(self, aligned_timestamps: Union[np.ndarray, List[np.ndarray
         aligned_timestamps : numpy.ndarray or list of numpy.ndarray
             The synchronized timestamps for data in this interface.
         """
-        if self._number_of_segments == 1:
-            self.recording_extractor.set_times(times=aligned_timestamps)
-        else:
-            assert isinstance(
-                aligned_timestamps, list
-            ), "Recording has multiple segment! Please pass a list of timestamps to align each segment."
-            assert (
-                len(aligned_timestamps) == self._number_of_segments
-            ), f"The number of timestamp vectors ({len(aligned_timestamps)}) does not match the number of segments ({self._number_of_segments})!"
+        assert (
+            self._number_of_segments == 1
+        ), "This recording has multiple segments; please use 'align_segment_timestamps' instead."
 
-            for segment_index in range(self._number_of_segments):
-                self.recording_extractor.set_times(times=aligned_timestamps[segment_index], segment_index=segment_index)
+        self.recording_extractor.set_times(times=aligned_timestamps)
+
+    def align_segment_timestamps(self, aligned_timestamps: List[np.ndarray]):
```

Comment:
@h-mayorquin One last thing, what do you prefer between these two options

(a) keyword arguments are like shown here; `aligned_timestamps` for consistency with `align_timestamps` signature

(b) add 'segment' to the keyword name, as `aligned_segment_timestamps` for extra specificity

Same argument would be made about the `segment_starting_times` in the other function